wagtail/admin/checks.py

```diff
@@ -166,6 +166,72 @@ def check_panels_in_model(cls, context="model"):
     return errors
 
 
+@register("panels")
+def panel_type_check(app_configs, **kwargs):
+    from wagtail.models import get_page_models
+    from wagtail.snippets.models import get_snippet_models
+
+    errors = []
+
+    for cls in get_page_models():
+        errors += traverse_edit_handlers(cls.get_edit_handler())
+
+    for cls in get_snippet_models():
+        if hasattr(cls, "panels"):
+            for panel in cls.panels:
+                # Check if panel is MultiFieldPanel
+                if hasattr(panel, "children"):
+                    for panel_child in panel.children:
+                        errors += traverse_edit_handlers(panel_child.bind_to_model(cls))
+                errors += traverse_edit_handlers(panel.bind_to_model(cls))
+
+    return errors
+
+
+def traverse_edit_handlers(edit_handler):
+    errors = []
+
+    try:
+        for child in edit_handler.children:
+            errors += traverse_edit_handlers(child)
+    except AttributeError:
+        panel_field_error = check_panel_field_valid(edit_handler)
+        if panel_field_error:
+            errors.append(panel_field_error)
+
+    return errors
+
+
+def check_panel_field_valid(edit_handler):
+    from wagtail.admin.panels import CommentPanel, TabbedInterface
+
+    if isinstance(edit_handler, (CommentPanel, TabbedInterface)):
+        pass
+    else:
+        model = getattr(edit_handler, "model", None)
+        field_or_relation_name = getattr(edit_handler, "field_name", None)
+        if field_or_relation_name is None:
+            field_or_relation_name = getattr(edit_handler, "relation_name", None)
+
+        if field_or_relation_name in [f.name for f in model._meta.get_fields()]:
+            pass
+        elif hasattr(edit_handler, "relation_name") and field_or_relation_name in [
+            getattr(f, "related_name", None) for f in model._meta.get_fields()
+        ]:
+            pass
+        else:
+            return Warning(
+                "{model} does not have a field named '{field_name}', but a {edit_handler} is pointing to it".format(
+                    model=model.__name__,
+                    field_name=field_or_relation_name,
+                    edit_handler=edit_handler.__class__.__name__,
+                ),
+                hint="Check whether the field exists on the model and ensure it has been typed correctly",
```

Comment:
```suggestion
                hint="Check whether the field exists on the model or whether there is a spelling issue.",
```

Just re-worded this, `typed correctly` may be confused with Python types.